internal/appsec/remoteconfig.go

```diff
@@ -165,6 +166,33 @@ func (a *appsec) onRCRulesUpdate(updates map[string]remoteconfig.ProductUpdate)
 			statuses[k] = genApplyStatus(true, err)
 		}
 	} else {
+		wafDiags := a.wafHandle.Diagnostics()
+		for field, entry := range map[string]*waf.DiagnosticEntry{
+			"rules":           wafDiags.Rules,
+			"custom_rules":    wafDiags.CustomRules,
+			"exclusions":      wafDiags.Exclusions,
+			"rules_overrides": wafDiags.RulesOverrides,
+			"rules_data":      wafDiags.RulesData,
+			"processors":      wafDiags.Processors,
+			"scanners":        wafDiags.Scanners,
+		} {
+			if entry == nil {
+				continue
+			}
+			if entry.Error != "" {
+				for k := range statuses {
+					statuses[k] = genApplyStatus(true, fmt.Errorf("the WAF rejected invalid %s: %s", field, entry.Error))
+				}
+				break // We are reporting failure, bail out now...
+			}
+			if len(entry.Failed) > 0 {
+				for k := range statuses {
+					statuses[k] = genApplyStatus(true, fmt.Errorf("the WAF failed to load some %s: %#v", field, entry.Failed))
+				}
+				break // We are reporting failure, bail out now...
+			}
+		}
+
```

Comment:
This block raised tons of questions for me that I discussed with Anil and we cannot report rules diagnostics via RC's apply status. They can return things upon new RC rules requiring later libddwaf versions, for instance when a new operator is introduced. 

So you must instead log them out as debug information only for now, and in the future (or now?) send them via telemetry logs. I'd suggest to hide this deeper insider `swapWAF()` instead.

This also means we should make sure `swapWAF()` only returns an error when the rule truly wasn't loaded and ended up unusable (I'd say in the case where the returned WAF handle is nil).

cc @Hellzy

Reply:
For the record, here is the telemetry logs request https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/producing-telemetry.md#logs - not so hard to add to the client (without wiring it to the tracer logger for now)

Reply:
Won't do the telemetry since it's currently unsupported (per the very doc that's linked).